app/src/main/java/org/simple/clinic/teleconsultlog/shareprescription/TeleconsultSharePrescriptionModel.kt

```diff
@@ -13,16 +13,18 @@ data class TeleconsultSharePrescriptionModel(
     val prescriptionDate: LocalDate,
     val medicines: List<PrescribedDrug>?,
     val medicalRegistrationId: String?,
-    val patientProfile: PatientProfile?
+    val patientProfile: PatientProfile?,
+    val downloadButtonState: ButtonState?
 ) : Parcelable {
 
   companion object {
-    fun create(patientUuid: UUID, prescriptionDate: LocalDate) = TeleconsultSharePrescriptionModel(
+    fun create(patientUuid: UUID, prescriptionDate: LocalDate, downloadButtonState: ButtonState?) = TeleconsultSharePrescriptionModel(
```

Comment:
Why are we passing this download button state to this method? The default state will be `null` and we hide the progress state for that anyway.

Reply:
We're adding the default state as `NOT_DOWNLOADING` and not null.

Reply:
We made the button state nullable in the model. Even if it's `null` we will hide the progress state since in the ui renderer we are only showing progress when the button state is `DOWNLOADING`. So I don't see a point of passing a button state to the factory method 🤔